browser/brave_rewards/rewards_service_factory.cc

```diff
@@ -83,9 +85,26 @@ RewardsServiceFactory::~RewardsServiceFactory() {
 KeyedService* RewardsServiceFactory::BuildServiceInstanceFor(
     content::BrowserContext* context) const {
 #if BUILDFLAG(BRAVE_REWARDS_ENABLED)
+  std::unique_ptr<RewardsServiceObserver> extension_observer = nullptr;
+  std::unique_ptr<RewardsServicePrivateObserver> private_observer = nullptr;
+  std::unique_ptr<RewardsNotificationServiceObserver> notification_observer =
+      nullptr;
+  #if BUILDFLAG(ENABLE_EXTENSIONS)
+    extension_observer = std::make_unique<ExtensionRewardsServiceObserver>(
+          Profile::FromBrowserContext(context));
+    private_observer = std::make_unique<ExtensionRewardsServiceObserver>(
+          Profile::FromBrowserContext(context));
+    notification_observer =
+        std::make_unique<ExtensionRewardsNotificationServiceObserver>(
+          Profile::FromBrowserContext(context));
+  #endif
+
   std::unique_ptr<RewardsServiceImpl> rewards_service(
       new RewardsServiceImpl(Profile::FromBrowserContext(context)));
-  rewards_service->Init();
+  rewards_service->Init(
```

Comment:
I would personally dump this method and pass everything into the constructor, but just a suggestion